### ISSUE TITLE
Is there a way to restart nodejs in a clean way

### ISSUE THREAD

**Issue by ingomueller-net** [AEGEE/oms-development-vm#16]
Currently nodejs is started by puppet. When modifications occur, the server needs to be restarted, however I can't think of any other way (except with puppet, which I am still not completely knowing) but to kill the process and restart it

What would you suggest?

----------------------------------------
- Bitbucket: https://bitbucket.org/ingomueller-net/ldap-poc/issue/16
- Originally reported by: [grasshopper](http://bitbucket.org/grasshopper)
- Originally created at: 2015-09-30T10:49:17.894

**Comment by ingomueller-net** [comment]
```
#!

sudo service oms-core restart
```

puppet is cool!

----------------------------------------
Original comment by: [grasshopper](http://bitbucket.org/grasshopper)

**Comment by ingomueller-net** [comment]
I am still not sure what is the best practice here... The upstart script (`oms-core.conf`) has nothing to do with puppet, but may be the way to go. It should probably be in the `oms-core` project though. (When I referred to puppet, I meant to restart the VM, which takes quite some time.)

----------------------------------------
Original comment by: Ingo Müller

**Comment by ingomueller-net** [comment]
I changed my mind: I think this is a good solution. In particular should `oms-core.conf` be in the puppet project, since it needs to know about where the code is etc -- things that oms-core should not know. oms-core just assumes that it is started somehow.

BTW, I verified that upstart+forever even monitor whether the server still runs, and restarts it in case in exited/crashed/was killed.

----------------------------------------
Original comment by: Ingo Müller
